src/main/java/net/spy/memcached/protocol/TCPMemcachedNodeImpl.java

```diff
@@ -731,6 +746,16 @@ public final void setupForAuth() {
     }
   }
 
+  @Override
+  public boolean executeTlsHandshake() {
+      try {
+        return tlsConnectionManager.doHandshake(channel);
```

Comment:
I don't think this currently handles the exception where `tlsConnectionManager` is null, right?

Reply:
I could go either way on if we need to handle null explicitly since this should never be called if ssl isn't enabled, so then if that happened and we threw a null pointer exception instead of an illegal state exception it's like 🤷‍♀️